src/main/java/nextstep/subway/domain/BaseEntity.java

```diff
@@ -0,0 +1,28 @@
+package nextstep.subway.domain;
+
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.*;
+import javax.swing.text.html.Option;
+import java.util.Optional;
+
+@MappedSuperclass
+public abstract class BaseEntity {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    public Long getId() {
+        return this.id;
+    }
+
+    public Optional<Long> getSafeId(){
+        return Optional.ofNullable(getId());
+    }
+
+    protected boolean isPersisted() {
+        return this.id != null;
+    }
```

Comment:
`SectionBuilder` 클래스에서 사용되고 있는데, 항상 `!`를 통해 반대로만 호출되고 있는 것 같아요.
이렇다면 차라리 메서드 자체를 반대 의미로 만들어보면 어떨까요?
```suggestion
    protected boolean isNotPersisted() {
        return this.id == null;
    }
```
```java
// SectionBuilder.java
// AS-IS
if (!this.upStation.isPersisted() || !this.downStation.isPersisted()) {
    throw new IllegalArgumentException("저장되지 않은 역입니다.");
}

// TO-BE
if (this.upStation.isNotPersisted() || this.downStation.isNotPersisted()) {
    throw new IllegalArgumentException("저장되지 않은 역입니다.");
}